WordPress/Classes/Models/AbstractPost.m

```diff
@@ -452,8 +452,17 @@ - (BOOL)isDraft
 
 - (void)publishImmediately
 {
-    self.dateModified = [NSDate date];
-    [self setDateCreated:self.dateModified];
+    // If the post was modified later than it was created
+    // then this is a post that was published at some point
+    // we will not update the dateCreated
+    if (self.dateCreated &&
+        self.dateModified &&
+        [self.dateCreated compare:self.dateModified] == NSOrderedAscending) {
```

Comment:
Maybe compose the check with `earlierDate` or `laterDate` instead of `compare` for extra clarity?